build-logic/src/main/kotlin/org/jetbrains/kotlin/gradle/internal/testing/TCServiceMessagesClient.kt

```diff
@@ -0,0 +1,554 @@
+// Copied from https://github.com/JetBrains/kotlin/blob/v1.7.21/libraries/tools/kotlin-gradle-plugin/src/common/kotlin/org/jetbrains/kotlin/gradle/internal/testing/TCServiceMessagesClient.kt
+// Modified to:
+// - remove all kotlinDebug and LegacyTestDescriptorInternal references that are internal to the Kotlin Gradle Plugin
+// - call failure() with a TestFailure instead of a Throwable
+// https://youtrack.jetbrains.com/issue/KT-54634
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.gradle.internal.testing
+
+import jetbrains.buildServer.messages.serviceMessages.*
+import org.gradle.api.internal.tasks.testing.*
+import org.gradle.api.tasks.testing.TestFailure
+import org.gradle.api.tasks.testing.TestOutputEvent
+import org.gradle.api.tasks.testing.TestOutputEvent.Destination.StdErr
+import org.gradle.api.tasks.testing.TestOutputEvent.Destination.StdOut
+import org.gradle.api.tasks.testing.TestResult
+import org.gradle.api.tasks.testing.TestResult.ResultType.*
+import org.gradle.internal.operations.OperationIdentifier
+import org.gradle.process.internal.ExecHandle
+import org.jetbrains.kotlin.gradle.internal.LogType
+import org.jetbrains.kotlin.gradle.testing.KotlinTestFailure
+import org.slf4j.Logger
+import java.text.ParseException
+
+data class TCServiceMessagesClientSettings(
```

Comment:
This is an overshoot of https://github.com/apollographql/apollo-kotlin/pull/4552. See https://youtrack.jetbrains.com/issue/KT-54634/MPP-Test-Failure-causes-KotlinJvmTestExecutorexecute1-does-not-define-failure#focus=Comments-27-6696205.0-0